models/user.js

```diff
@@ -310,56 +312,84 @@ UserService.createLocalUsers = (users) => {
 };
 
 /**
+ * Check the requested displayname for naughty words (currently in English) and special chars
+ * @param  {String}   displayName word to be checked for profanity
+ * @return {Promise}  rejected if the machine's sensibilites are offended
+ */
+const isValidDisplayName = (displayName) => {
+  const onlyLettersNumbersUnderscore = /^[a-z0-9_]+$/;
+
+  if (!displayName) {
+    return Promise.reject(errors.ErrMissingDisplay);
+  }
+
+  if (!onlyLettersNumbersUnderscore.test(displayName)) {
+
+    return Promise.reject(errors.ErrSpecialChars);
+  }
+
+  // check for profanity
+  return Wordlist.displayNameCheck(displayName);
+};
+
+/**
  * Creates the local user with a given email, password, and name.
  * @param  {String}   email       email of the new user
  * @param  {String}   password    plaintext password of the new user
  * @param  {String}   displayName name of the display user
  * @param  {Function} done        callback
  */
 UserService.createLocalUser = (email, password, displayName) => {
+
   if (!email) {
-    return Promise.reject('email is required');
+
+    return Promise.reject(errors.ErrMissingEmail);
   }
 
-  email = email.toLowerCase();
+  email = email.toLowerCase().trim();
+  displayName = displayName.toLowerCase().trim();
```

Comment:
I think that these transformations should occur on the handler of the endpoints and not as a part of the service..